src/modules/credentials/handlers/ProposeCredentialHandler.ts

```diff
@@ -1,17 +1,76 @@
+import type { AgentConfig } from '../../../agent/AgentConfig'
 import type { Handler, HandlerInboundMessage } from '../../../agent/Handler'
+import type { CredentialRecord } from '../repository/CredentialRecord'
 import type { CredentialService } from '../services'
 
+import { createOutboundMessage } from '../../../agent/helpers'
+import { AriesFrameworkError } from '../../../error'
+import { AutoAcceptCredential } from '../../../types'
+import { CredentialUtils } from '../CredentialUtils'
 import { ProposeCredentialMessage } from '../messages'
 
 export class ProposeCredentialHandler implements Handler {
   private credentialService: CredentialService
+  private agentConfig: AgentConfig
   public supportedMessages = [ProposeCredentialMessage]
 
-  public constructor(credentialService: CredentialService) {
+  public constructor(credentialService: CredentialService, agentConfig: AgentConfig) {
     this.credentialService = credentialService
+    this.agentConfig = agentConfig
   }
 
   public async handle(messageContext: HandlerInboundMessage<ProposeCredentialHandler>) {
-    await this.credentialService.processProposal(messageContext)
+    const credentialRecord = await this.credentialService.processProposal(messageContext)
+
+    const autoAccept = CredentialUtils.composeAutoAccept(
+      credentialRecord.autoAcceptCredential,
+      this.agentConfig.autoAcceptCredentials
+    )
+
+    if (autoAccept === AutoAcceptCredential.always) {
```

Comment:
I think it would be nice if we can extract all this logic into an auto respond handler. So something like `xxxx.shouldAutoRespondToProposal()`

Reply:
This way we can reuse some of the logic and have a single place to handle this.